tests/test.js

```diff
@@ -8,8 +8,9 @@ var server = connect().use(serveStatic(path.join(__dirname, '/..'))).listen(8071
 	if (process.argv.indexOf('--phantomjs') !== -1) {
 		childProcess.spawn('node_modules/.bin/mocha-phantomjs', ['http://localhost:8071/tests/tests.html'], {
 			stdio: 'inherit'
-		}).on('exit', function() {
+		}).on('exit', function(code) {
```

Comment:
Why is this required? It seems not related to the PR

Reply:
See https://travis-ci.org/mholt/PapaParse/jobs/389825011#L1418

While working on this I commited a bad code and the tests were still green.

So I have updated this to see the tests failing and then fixed the code.

I tried to document it in "Changes"

```
As a drive-by fix, I have updated the phantomjs test runner to propagate the exit code
as it was always exiting with 0 even on errors see https://travis-ci.org/mholt/PapaParse/jobs/389825011#L1418
```

Let me know if the PR description does not make sense and I will try to repharse it.

Thanks!

Reply:
Uops, this sounds as a separate issue which scares me a little bit. 
Could you please include this CHANGE in a separate PR? I will like to merge it first.

Reply:
Thanks a lot for the review.

I am not a day to say JS developers so I don't see why this change is so dangerous. 
Why it scares you?

For me, this PR has all the parts for this change and the tests are green.

There is a commit in which we see that the tests are green even with broken code.
There is a separate commit in which the infrastructure code is fixed.
Then we have another commit with bad code which pass on node but fails on phantomjs and for this we can see that Travis-CI is red.
And then a final commit where the phantomjs code is fixed and Travis-CI is fixed.

Putting this into a separate PR would mean that I would have to invent bogus code to have to test break, and then fix the test runner then fix the code... it feels like a lot of extra work.

This is  Boyscout rule https://deviq.com/boy-scout-rule/.

I think that if you put too much extra red tape on code cleanup you might end up with discouraging the cleanup.

I understand the purpose of small PR and a single change in the PR, but I don't think that they should apply to the testing infrastructure code.
I think that you should be allowed to do multiple changes to the infrastrucrure in order to best server the targeted change.

Let me know if you still want this change in a separate and I will do it :)

Thanks again for the review.

Reply:
It scares me that we can find some error that was silently hidden by this code.
Just include the same change in a separate PR, there is no need to add something else. 

I prefer to have a separate PR to keep history cleaner by not include two unrelated changes in the same commit as I normally squash the PR into a single commit before merging.

Reply:
Done https://github.com/mholt/PapaParse/pull/523

Well, it can happen. There is no test for the test infrastructure so it relies on people writing bad code accidentally (like in my case) or on purpose (if you are religious about tdd )